src/paths/dijkstra.c

```diff
@@ -948,27 +970,56 @@ int igraph_get_all_shortest_paths_dijkstra(const igraph_t *graph,
         n = igraph_vector_size(&order);
         igraph_vector_null(paths_index);
 
-        /* clear the paths vector */
-        igraph_vector_ptr_clear(res);
-        igraph_vector_ptr_set_item_destructor(res,
-                                              (igraph_finally_func_t*)igraph_vector_destroy);
-
         /* by definition, the shortest path leading to the starting vertex
-         * consists of the vertex itself only */
+        * consists of the vertex itself only */
         path = igraph_Calloc(1, igraph_vector_t);
         if (path == 0)
             IGRAPH_ERROR("cannot run igraph_get_all_shortest_paths_dijkstra",
-                         IGRAPH_ENOMEM);
+                        IGRAPH_ENOMEM);
         IGRAPH_FINALLY(igraph_free, path);
         IGRAPH_CHECK(igraph_vector_init(path, 1));
-        IGRAPH_CHECK(igraph_vector_ptr_push_back(res, path));
+
+        /* clear the paths vector */
+        /* If res isn't excist then create one, in order for the algorithm to work. */
+        if (!res){
+            res = igraph_Calloc(1, igraph_vector_ptr_t);
+            if (res == 0) {
+                IGRAPH_ERROR("cannot run igraph_get_all_shortest_paths", IGRAPH_ENOMEM);
+            }
+            IGRAPH_CHECK(igraph_vector_ptr_init(res, 0));
+            igraph_vector_ptr_set_item_destructor(res, (igraph_finally_func_t*)igraph_vector_destroy);
+            free_res = 1;
+            IGRAPH_CHECK(igraph_vector_ptr_push_back(res, path));
+        }
+        else{
+            igraph_vector_ptr_clear(res);
+            igraph_vector_ptr_set_item_destructor(res,
+                                            (igraph_finally_func_t*)igraph_vector_destroy);
+            IGRAPH_CHECK(igraph_vector_ptr_push_back(res, path));
+        }
+        
         IGRAPH_FINALLY_CLEAN(1);  /* ownership of path passed to res */
         VECTOR(*path)[0] = from;
+        if (res_e){
+            igraph_vector_ptr_clear(res_e);
+            igraph_vector_ptr_set_item_destructor(res_e,
+                                              (igraph_finally_func_t*)igraph_vector_destroy);
+            path = igraph_Calloc(1, igraph_vector_t);
+            if (path == 0)
+                IGRAPH_ERROR("cannot run igraph_get_all_shortest_paths_dijkstra",
+                            IGRAPH_ENOMEM);
+            /* the shortest path from the source to itself consist of the edge -1 */
+            IGRAPH_FINALLY(igraph_free, path);
+            IGRAPH_CHECK(igraph_vector_init(path, 1));
+            IGRAPH_CHECK(igraph_vector_ptr_push_back(res_e, path));
+            IGRAPH_FINALLY_CLEAN(1);  /* ownership of path passed to res */
+            VECTOR(*path)[0] = -1;
```

Comment:
Not sure if I like this -1 in the edge ID vector; couldn't we simply keep the vector empty?

Reply:
I prefer to keep it that way because path is initialized to 1.
Is this O.k?

Reply:
See my comment above -- can't we simply call `igraph_vector_init(path, 0)` to create an empty vector?